sklearn/neighbors/classification.py

```diff
@@ -155,18 +156,37 @@ def predict(self, X):
         weights = _get_weights(neigh_dist, self.weights)
 
         y_pred = np.empty((n_samples, n_outputs), dtype=classes_[0].dtype)
+
+        if self._issparse:
+            y_pred_sparse = []
+
         for k, classes_k in enumerate(classes_):
-            if weights is None:
-                mode, _ = stats.mode(_y[neigh_ind, k], axis=1)
+
+            if self._issparse and self.outputs_2d_:
+                if weights is None:
+                    mode, _ = stats.mode(_y[neigh_ind, k].toarray(), axis=1)
```

Comment:
It might be interesting to implement a sparse (weighted) mode, but this is fine for now. Requires memory `O(n_samples x n_neighbors)` where sparse is being used to avoid memory `O(n_samples x n_outputs)`